mesonbuild/build.py

```diff
@@ -724,7 +724,7 @@ def __init__(
             objects: T.List[ObjectTypes],
             environment: environment.Environment,
             compilers: T.Dict[str, 'Compiler'],
-            kwargs):
+            kwargs: T.Dict[str, T.Any]):
```

Comment:
Couldn't we use TYPE_kwargs here?

Reply:
Well, it is narrower than Any which explicitly allows anything (and requires that you support all generic objects, too, it's not like a type ignore)